src/lightning_app/cli/commands/app_commands.py

```diff
@@ -5,40 +5,47 @@
 import requests
 
 from lightning_app.cli.commands.connection import _resolve_command_path
-from lightning_app.utilities.cli_helpers import _retrieve_application_url_and_available_commands
+from lightning_app.utilities.cli_helpers import _LightningAppOpenAPIRetriever
 from lightning_app.utilities.commands.base import _download_command
 from lightning_app.utilities.enum import OpenAPITags
 
 
 def _run_app_command(app_name: str, app_id: Optional[str]):
     """Execute a function in a running App from its name."""
     # 1: Collect the url and comments from the running application
-    url, api_commands, _ = _retrieve_application_url_and_available_commands(app_id)
-    if url is None or api_commands is None:
-        raise Exception("We couldn't find any matching running App.")
+    running_help = sys.argv[-1] == "--help"
 
-    if not api_commands:
+    retriever = _LightningAppOpenAPIRetriever(app_id, use_cache=running_help)
+
+    if not running_help and (retriever.url is None or retriever.api_commands is None):
+        if app_name == "localhost":
+            print("The command couldn't be executed as your local Lightning App isn't running.")
```

Comment:
Any reason for print instead of logging?

Reply:
This is user facing. You want print instead of info.